mlflow/sklearn/__init__.py

```diff
@@ -702,145 +691,6 @@ def fit_mlflow(original, self, *args, **kwargs):
         _log_posttraining_metadata(self, *args, **kwargs)
         return fit_output
 
-    def _log_pretraining_metadata(estimator, *args, **kwargs):  # pylint: disable=unused-argument
-        """
-        Records metadata (e.g., params and tags) for a scikit-learn estimator prior to training.
-        This is intended to be invoked within a patched scikit-learn training routine
-        (e.g., `fit()`, `fit_transform()`, ...) and assumes the existence of an active
-        MLflow run that can be referenced via the fluent Tracking API.
-
-        :param estimator: The scikit-learn estimator for which to log metadata.
-        :param args: The arguments passed to the scikit-learn training routine (e.g.,
-                     `fit()`, `fit_transform()`, ...).
-        :param kwargs: The keyword arguments passed to the scikit-learn training routine.
-        """
-        # Deep parameter logging includes parameters from children of a given
-        # estimator. For some meta estimators (e.g., pipelines), recording
-        # these parameters is desirable. For parameter search estimators,
-        # however, child estimators act as seeds for the parameter search
-        # process; accordingly, we avoid logging initial, untuned parameters
-        # for these seed estimators.
-        should_log_params_deeply = not _is_parameter_search_estimator(estimator)
-        # Chunk model parameters to avoid hitting the log_batch API limit
-        for chunk in _chunk_dict(
-            estimator.get_params(deep=should_log_params_deeply),
-            chunk_size=MAX_PARAMS_TAGS_PER_BATCH,
-        ):
-            truncated = _truncate_dict(chunk, MAX_ENTITY_KEY_LENGTH, MAX_PARAM_VAL_LENGTH)
-            try_mlflow_log(mlflow.log_params, truncated)
-
-        try_mlflow_log(mlflow.set_tags, _get_estimator_info_tags(estimator))
-
-    def _log_posttraining_metadata(estimator, *args, **kwargs):
```

Comment:
Move these into `utils` for testability. We currently have tests (e.g., `test_fit_xxx_performs_logging_only_once`)  verifying that `log_metrics` / `log_params`/ `set_tags` are called several times; these tests are brittle and hard to interpret, especially as we change which APIs we're using to log in an effort to batch information.

This PR updates the tests to instead verify that `_log_pretraining_metadata` and `_log_posttraining_metadata` are called once.